### PR TITLE
docs: Update project structure and design

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Updates to the docs, including,

* Project Structure
  * Minor typos on project structure
  * Some rewording for clarity
  * Remove duplicate listings of `test-utils` and `test-utils-core
* Design
  * Figma is no longer a "future" thing (I beileve).
  * Linked milestones and issues.
* Snuck in a header fix for `__mock__` I had laying around in a stash

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4565-z2fiyjdu5efdha8kah8t0fwazagzsndv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602de2f5f285af4858919657

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4565-z2fiyjdu5efdha8kah8t0fwazagzsndv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602de2f5f285af4858919657

File: docs/support/project-structure.md [hunk 1]
```diff
@@ -196,10 +193,10 @@ We can categorize plugins into three different types; **Frontend**, **Backend**
 and **GraphQL**. We differentiate these types of plugins when we name them, with
 a dash-suffix. `-backend` means it’s a backend plugin and so on.
 
-One reason for splitting a plugin is because of to it's dependencies. Another
+One reason for splitting a plugin is because of it's dependencies. Another
```
Comment (freben) [inline docs/support/project-structure.md:196]:
```suggestion
One reason for splitting a plugin is because of its dependencies. Another
```

**Review by freben — APPROVED** [review]
One small comment that wasn't even your fault. Good job!

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4565-z2fiyjdu5efdha8kah8t0fwazagzsndv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602de2f5f285af4858919657

**Pull request merged**